qiskit_experiments/tomography/tomography_experiment.py

```diff
@@ -0,0 +1,307 @@
+
+            if prep_element:
+                # Add tomography preparation
+                prep_qubits = self._prep_qubits or range(self.num_qubits)
+                prep_circ = self._prep_circ_basis.circuit(prep_element)
+                circ.reset(prep_qubits)
+                circ.append(prep_circ, prep_qubits)
+                circ.barrier(prep_qubits)
+
+            # Add target circuit
+            # Have to use compose since circuit.to_instruction has a bug
+            # when circuit contains classical registers and conditionals
+            circ = circ.compose(self._circuit, circ_qubits, circ_clbits)
+
+            # Add tomography measurement
+            meas_circ = self._meas_circ_basis.circuit(meas_element)
+            circ.barrier(meas_qubits)
+            circ.append(meas_circ, meas_qubits)
+            circ.measure(meas_qubits, meas_clbits)
+
+            # Add metadata
+            circ.metadata = metadata
+            circuits.append(circ)
+        return circuits
+
+    def _basis_indices(self):
+        """Return list of basis element indices"""
+        basis_indices = self.experiment_options.basis_indices
+        if basis_indices is not None:
+            return basis_indices
+
+        meas_size = len(self._meas_circ_basis)
+        num_meas = len(self._meas_qubits) if self._meas_qubits else self.num_qubits
+        meas_elements = product(range(meas_size), repeat=num_meas)
+        if self._prep_circ_basis:
+            prep_size = len(self._prep_circ_basis)
+            num_prep = len(self._prep_qubits) if self._prep_qubits else self.num_qubits
+            prep_elements = product(range(prep_size), repeat=num_prep)
+        else:
+            prep_elements = [None]
+
+        return product(meas_elements, prep_elements)
+
+    @staticmethod
+    def _permute_circuit(
+        circuit: QuantumCircuit,
+        measurement_qubits: Optional[Iterable[int]] = None,
+        preparation_qubits: Optional[Iterable[int]] = None,
+    ):
+        """Permute circuit qubits.
+
+        This permutes the circuit so that the specified preparation and measurement
+        qubits correspond to input and output qubits [0, ..., N-1] respectively
+        for the returned circuit.
+        """
+        if measurement_qubits is None and preparation_qubits is None:
+            return circuit
+
+        total_qubits = circuit.num_qubits
+        total_clbits = circuit.num_clbits
+        if total_clbits:
+            perm_circ = QuantumCircuit(total_qubits, total_clbits)
+        else:
+            perm_circ = QuantumCircuit(total_qubits)
+
+        # Apply permutation to put prep qubits as [0, ..., M-1]
+        if preparation_qubits:
+            prep_qargs = list(preparation_qubits)
+            if len(preparation_qubits) != total_qubits:
+                prep_qargs += [i for i in range(total_qubits) if i not in preparation_qubits]
+            perm_circ.append(Permutation(total_qubits, prep_qargs).inverse(), range(total_qubits))
```

Comment:
This function is only used for computing the target state/channel and is the most robust way to do without having to do expensive state/channel class dependent array manipulations. Preparation and measurement qubits are order sensitive. State tomography of measurement_qubits [0, 1] and [1,0] will have qubits reversed in the fitted density matrix.

Your example should raise an exception at class init, because qubit 5 isn't in the input circuit, so i need to add that validation.